packages/svg-icons/scripts/changeColorPlugin.ts

```diff
@@ -0,0 +1,35 @@
+import type { XastElement } from "svgo/lib/types.js";
+
+import { colors } from "./svgoConfig.ts";
+
+const colorsProps = new Set([
+    "color",
+    "fill",
+    "flood-color",
+    "lighting-color",
+    "stop-color",
+    "stroke"
+]);
+
+export const changeColorPlugin = {
+    name: "changeColorPlugin",
+    description: "Change the color of the SVGs",
+    fn: () => {
+        return {
+            element: {
+                enter: (node: XastElement) => {
+                    for (const [nodeName, nodeValue] of Object.entries(node.attributes)) {
+                        if (colorsProps.has(nodeName)) {
+                            let value = nodeValue as string;
+
+                            if (colors[value]) {
+                                value = colors[value];
+                            }
+                            node.attributes[nodeName] = value;
+                        }
+                    }
+                }
+            }
+        };
+    }
+};
```

Comment:
Now that the conversion has been made in the svg-icons step, i think we can remove some code in the icons package. 

 ```
                replaceAttrValues: {
                    [PrimaryIconColor]: "var(--hop-primary-icon)",
                    [WarningWeakIconColor]: "var(--hop-warning-icon-weak)",
                    [DecorativeOption7IconColor]: `var(--hop-richicon-placeholder-icon-color, ${DecorativeOption7IconColor})`,
                    [DecorativeOption7SurfaceColor]: `var(--hop-richicon-placeholder-surface-color, ${DecorativeOption7SurfaceColor})`,
                    [White]: `var(--hop-richicon-placeholder-icon-strong-color, ${White})` // TODO: should be --hop-decorative-option7-icon-strong once it exists
                },
```

we can remove those 3 lines: 
```
[DecorativeOption7IconColor]: `var(--hop-richicon-placeholder-icon-color, ${DecorativeOption7IconColor})`,
                    [DecorativeOption7SurfaceColor]: `var(--hop-richicon-placeholder-surface-color, ${DecorativeOption7SurfaceColor})`,
                    [White]: `var(--hop-richicon-placeholder-icon-strong-color, ${White})` // TODO: should be --hop-decorative-option7-icon-strong once it exists
```

Reply:
Nous devrions également modifier les variable dans icons packages 

```
replaceAttrValues: {
      [PrimaryIconColor]: "var(--hop-primary-icon)",
      [WarningWeakIconColor]: "var(--hop-warning-icon-weak)"
  },
```

```
replaceAttrValues: {
      [PrimaryIconColor]: "var(--hop-Icon-primary-icon)",
      [WarningWeakIconColor]: "var(--hop-Icon-warning-icon-weak)"
  },
```

Reply:
En fait ça pourrait même être fait directement dans le package svg-icons, avec la même technique que tu as présentement? comme ça les gens pourrait modifié les couleurs des icones dans svg-icons aussi